examples/ota-requestor-app/linux/main.cpp

```diff
@@ -47,33 +52,40 @@ using chip::Server;
 using chip::VendorId;
 using chip::bdx::TransferSession;
 using chip::Callback::Callback;
-using chip::Inet::IPAddress;
 using chip::System::Layer;
 using chip::Transport::PeerAddress;
 using namespace chip::ArgParser;
 using namespace chip::Messaging;
 using namespace chip::app::Clusters::OtaSoftwareUpdateProvider::Commands;
 
+enum OnConnectedState
+{
+    kQueryImage = 0,
+    kStartBDX,
+};
+
 void OnQueryImageResponse(void * context, const QueryImageResponse::DecodableType & response);
 void OnQueryImageFailure(void * context, EmberAfStatus status);
 void OnConnected(void * context, chip::DeviceProxy * deviceProxy);
 void OnConnectionFailure(void * context, NodeId deviceId, CHIP_ERROR error);
 bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier, const char * aName, const char * aValue);
 
 // TODO: would be nicer to encapsulate these globals and the callbacks in some sort of class
-ExchangeContext * exchangeCtx = nullptr;
+CASESessionManager * mCASESessionManager = nullptr;
+OnConnectedState onConnectedState        = kQueryImage;
+ExchangeContext * exchangeCtx            = nullptr;
 BdxDownloader bdxDownloader;
 Callback<OnDeviceConnected> mOnConnectedCallback(OnConnected, nullptr);
 Callback<OnDeviceConnectionFailure> mOnConnectionFailureCallback(OnConnectionFailure, nullptr);
```

Comment:
Definitely, since more and more global state is added, it would be good to encapsulate it now. The threshold I think has now been met.

Also, some of the variables here (e.g. `mCASESesionManager`) and `mOnConnectedCallback`) use the "member variable" convention (`m` prefix), but are actually globals. This makes it hard further down the code to understand the scoping.

Reply:
@selissia is actively moving all this common logic out to OTARequestor class. I didn't want to duplicate the effort but understand it's starting to look messy.

Reply:
Also, with regards to the "m" convention, I was thinking these would be member variables when they are encapsulated but if it's confusing, I can remove the "m" for now until the refactor by Sergei is done.